### DISCUSSION TITLE
Python packages and pip

### DISCUSSION THREAD

**Discussion by PhilippImhof** [Homebrew/discussions#309] [Everyday usage]
I have just recently switched from .... another package manage to brew. In this context, I noticed a minor difference w.r.t. Python packages: If I run `brew install flake8` or  `brew install pylint`, the corresponding tools are installed and available as executables `/usr/local/bin/flake8` or `/usr/local/bin/pylint`.

However I cannot invoke as Python modules:
```
$ python3 -m flake8
/usr/local/opt/python@3.9/bin/python3.9: No module named flake8
```

This is different e.g. for `pip` which can be invoked as `pip` or as `python3 -m pip`. As a consequence, some tools have to be reconfigured, because their default is to just call `python3` with the corresponding module. 

There is probably a good reason why things are set up the way they are. Would anyone mind explaining me the details?

**Comment by jonchang** [comment]
Our documentation covers this. See the introductory paragraphs in [Python for Formula Authors](https://docs.brew.sh/Python-for-Formula-Authors).
